core/templates/pages/classroom-admin-page/classroom-admin-page.component.spec.ts

```diff
@@ -179,16 +234,49 @@ describe('Classroom Admin Page component ', () => {
     expect(component.classroomViewerMode).toBeFalse();
   }));
 
+  it('should set topicsToClassroomRelation and filteredTopicsToClassroomRelation', fakeAsync(() => {
+    const response = [
+      {
+        topic_name: 'topic_name',
+        topic_id: 'topic_id',
+        classroom_name: null,
+        classroom_url_fragment: null,
+      },
+    ];
+    spyOn(
+      classroomBackendApiService,
+      'getAllTopicsToClassroomRelation'
+    ).and.returnValue(Promise.resolve(response));
+
+    component.getClassroomData('getAllTopicsToClassroomRelation');
+    tick();
+
+    expect(component.topicsToClassroomRelation.length).toEqual(1);
+    expect(component.filteredTopicsToClassroomRelation.length).toEqual(1);
```

Comment:
Should we add a pre-check for this?

Reply:
can you please explain a little more about this?

Reply:
Should we check the initial values before the "getClassroomData call?

Reply:
Done. Added check before the getClassroomData call the topicsToClassroomRelation array should be empty.